### PR TITLE
verification & notfound errors

### PR DESCRIPTION
**Pull request by abdulgalimov**

### CODE REVIEW

**Review by AnnaNX1 — APPROVED** [review]

File: packages/substrate-client/tokens/methods/destroy-collection/method.ts [hunk 1]
```diff
@@ -20,12 +21,15 @@ export class DestroyCollection extends MutationMethodBase<
...
+      throw new NotFoundError('Not found collection', {
+        collectionId,
+      });
 
-    if (!collection.limits?.ownerCanDestroy)
+    // todo: think about how best to check limits, e.g. load effective limits
+    if (
+      !collection.limits?.ownerCanDestroy &&
+      typeof collection.limits?.ownerCanDestroy === 'boolean'
+    )
```
Comment (pasynkov) [inline packages/substrate-client/tokens/methods/destroy-collection/method.ts:32]: `{`

File: packages/substrate-client/tokens/methods/destroy-collection/method.ts [hunk 2]
```diff
@@ -20,12 +21,15 @@ export class DestroyCollection extends MutationMethodBase<
     const collection = await this.sdk.collections.get_new({ collectionId });
 
     if (!collection)
-      return {
-        isAllowed: false,
-        message: 'Not found collection',
-      };
+      throw new NotFoundError('Not found collection', {
```
Comment (pasynkov) [inline packages/substrate-client/tokens/methods/destroy-collection/method.ts:24]: `{`

File: packages/substrate-client/tokens/methods/destroy-collection/method.ts (hunk 1 above)
Comment (abdulgalimov) [reply packages/substrate-client/tokens/methods/destroy-collection/method.ts]: done

File: packages/substrate-client/tokens/methods/destroy-collection/method.ts (hunk 2 above)
Comment (abdulgalimov) [reply packages/substrate-client/tokens/methods/destroy-collection/method.ts:24]: done

**Review by pasynkov — APPROVED** [review]

**Pull request merged**